config/validate_server_test.go

```diff
@@ -53,8 +54,13 @@ func TestServerConfig_Validate(t *testing.T) {
 		err := s.Validate()
 		assert.NoError(t, err)
 
-		assert.Equal(t, DefaultHTTPRequestIdleTimeout, s.IdleTimeout)
-		assert.Equal(t, DefaultHTTPRequestWriteTimeout, s.WriteTimeout)
-		assert.Equal(t, DefaultHTTPRequestReadTimeout, s.ReadTimeout)
+		idleTimeout, _ := time.ParseDuration(ServerIdleTimeoutDefault)
+		assert.Equal(t, idleTimeout, s.IdleTimeout)
+
+		writeTimeout, _ := time.ParseDuration(ServerWriteTimeoutDefault)
+		assert.Equal(t, writeTimeout, s.WriteTimeout)
+
+		readTimeout, _ := time.ParseDuration(ServerReadTimeoutDefault)
+		assert.Equal(t, readTimeout, s.ReadTimeout)
```

Comment:
This is showing in a bright light how this weird solution in ServerConfig.Validate is making a code confusing. What we're testing? That when we're setting values and making validation of them, we will end up with different values which are the default values 😵‍💫

Reply:
You're absolutely right. Changed.